tsdb/chunkenc/histo.go

```diff
@@ -249,42 +258,43 @@ func (a *HistoAppender) Append(int64, float64) {}
 // * any buckets disappeared
 // * there was a counter reset in the count of observations or in any bucket, including the zero bucket
 // * the last sample in the chunk was stale while the current sample is not stale
+// It returns an additional boolean set to true if it is not appendable because of a counter reset.
 // If the given sample is stale, it will always return true.
-func (a *HistoAppender) Appendable(h histogram.SparseHistogram) ([]Interjection, []Interjection, bool) {
+func (a *HistoAppender) Appendable(h histogram.SparseHistogram) ([]Interjection, []Interjection, bool, bool) {
 	if value.IsStaleNaN(h.Sum) {
 		// This is a stale sample whose buckets and spans don't matter.
-		return nil, nil, true
+		return nil, nil, true, false
 	}
 	if value.IsStaleNaN(a.sum) {
 		// If the last sample was stale, then we can only accept stale samples in this chunk.
-		return nil, nil, false
+		return nil, nil, false, false
 	}
 
 	if h.Schema != a.schema || h.ZeroThreshold != a.zeroThreshold {
-		return nil, nil, false
+		return nil, nil, false, false
 	}
 	posInterjections, ok := compareSpans(a.posSpans, h.PositiveSpans)
 	if !ok {
-		return nil, nil, false
+		return nil, nil, false, false
```

Comment:
I think in this case, we always have a counter reset because `compareSpans` returns `false` iff a bucket has disappeared. If I'm right, then we should return `nil, nil, false, true` here.